contracts/DixelClubV2NFT.sol

```diff
@@ -76,51 +92,52 @@ contract DixelClubV2NFT is ERC721Enumerable, Ownable, SVGGenerator {
         _mintNewEdition(owner_, palette_);
     }
 
-    function mint(address to, uint24[PALETTE_SIZE] memory palette) public payable {
-        uint256 mintingCost = _metaData.mintingCost;
+    function mint(address to, uint24[PALETTE_SIZE] calldata palette) public payable {
+        uint256 mintingCost = uint256(_metaData.mintingCost);
 
-        require(msg.value == mintingCost, "INVALID_MINTING_COST_SENT");
-        require(_tokenIdTracker.current() < _metaData.maxSupply, "MAX_SUPPLY_REACHED");
-        require(block.timestamp >= _metaData.mintingBeginsFrom, "MINTING_NOT_STARTED_YET");
+        if(msg.value != mintingCost) revert DixelClubV2__InvalidCost(mintingCost, msg.value);
+        if(_tokenIdTracker >= _metaData.maxSupply) revert DixelClubV2__MaximumMinted();
+        if(uint32(block.timestamp) < _metaData.mintingBeginsFrom) revert DixelClubV2__NotStarted(_metaData.mintingBeginsFrom, uint32(block.timestamp));
 
         // For whitelist only collections
-        if (_metaData.whitelistOnly) {
-            require(isWhitelistWallet(msg.sender), "NOT_IN_WTHIELIST");
-
-            _removeWhitelist(msg.sender); // decrease allowance by 1
-        }
+        if (_metaData.whitelistOnly) _removeWhitelist(msg.sender); // decrease allowance by 1
 
         if (mintingCost > 0) {
             // Send fee to the beneficiary
             uint256 fee = mintingCost * _factory.mintingFee() / FRICTION_BASE;
-            (bool sent, ) = (_factory.beneficiary()).call{ value: fee }("");
-            require(sent, "FEE_TRANSFER_FAILED");
+
+            (bool sent, ) = _factory.beneficiary().call{ value: fee }("");
+            require(sent);
 
             // Send the rest of minting cost to the collection creator
-            (bool sent2, ) = (owner()).call{ value: mintingCost - fee }("");
-            require(sent2, "MINTING_COST_TRANSFER_FAILED");
+            (sent, ) = owner().call{ value: mintingCost - fee }("");
+            require(sent);
         }
 
         _mintNewEdition(to, palette);
     }
 
     function burn(uint256 tokenId) external {
-        require(_isApprovedOrOwner(msg.sender, tokenId), "CALLER_IS_NOT_APPROVED"); // This will check existence of token
+        if(!_isApprovedOrOwner(msg.sender, tokenId)) revert DixelClubV2__NotApproved(); // This will check existence of token
 
         delete _editionData[tokenId];
         _burn(tokenId);
     }
 
-    function _mintNewEdition(address to, uint24[PALETTE_SIZE] memory palette) private {
+    function _mintNewEdition(address to, uint24[PALETTE_SIZE] calldata palette) private {
         // We cannot just use balanceOf to create the new tokenId because tokens
         // can be burned (destroyed), so we need a separate counter.
-        uint256 tokenId = _tokenIdTracker.current();
+        uint256 tokenId;
+        unchecked {
+            tokenId = _tokenIdTracker++;
```

Comment:
for the security,
leave that line, even if re-enterant, the token number is increased by one more.